### PR TITLE
Provided library components + documentation + related tests

### PR DESCRIPTION
**Pull request by ekkinox**
see https://oat-sa.atlassian.net/browse/NEX-367

### CODE REVIEW

File: src/Registry/CorrelationIdsRegistry.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        return $this->parentCorrelationId;
+    }
+
+    public function getRootCorrelationId(): ?string
+    {
+        return $this->rootCorrelationId;
+    }
+
+    public function update(CorrelationIdsRegistryInterface $registry): CorrelationIdsRegistryInterface
```
Comment (emgolubev) [inline src/Registry/CorrelationIdsRegistry.php:57]: Optional. Maybe `replace` is a more accurate name for this method. WDYT? Or this is the only example of implementation?

**Review by emgolubev — COMMENTED** [review]
Great job! I added one optional question then will approve.

File: src/Registry/CorrelationIdsRegistry.php (hunk 1 above)
Comment (ekkinox) [reply src/Registry/CorrelationIdsRegistry.php:57]: we update internals from an other registry, but we keep the same object instance at the end.

Maybe `replaceValues` could be more accurate, but imo `replace` is a bit misleading.

And as for usages, it'll be done on other library levels: monolog  / guzzle bridges

wdyt ?

Comment (emgolubev) [reply src/Registry/CorrelationIdsRegistry.php:57]: Let's keep `update`

**Review by emgolubev — APPROVED** [review]

**Pull request merged**